### PR TITLE
Defined entry point so other packages can add force field directories

### PR DESCRIPTION
**Pull request by peastman**
The group is called `openmm.forcefielddir`.  The entry point should be a function that returns a string containing the path to a directory.

See https://github.com/choderalab/openmm-forcefields/issues/6.  Also see http://setuptools.readthedocs.io/en/latest/setuptools.html#dynamic-discovery-of-services-and-plugins for information on creating entry points.

### CODE REVIEW

**Comment by jchodera** [comment]
Thanks! I'll give this a try!

File: wrappers/python/simtk/openmm/app/forcefield.py [hunk 1]
```diff
@@ -47,6 +47,16 @@
 from simtk.openmm.app import Topology
 from simtk.openmm.app.internal.singleton import Singleton
 
+# Directories from which to load built in force fields.
+
+_dataDirectories = [os.path.join(os.path.dirname(__file__), 'data')]
+try:
+    from pkg_resources import iter_entry_points
+    for entry in iter_entry_points(group='openmm.forcefielddir'):
```
Comment (jchodera) [inline wrappers/python/simtk/openmm/app/forcefield.py]: What about a less awkward name, like `openmm.forcefield_directory`?

**Comment by jchodera** [comment]
Actually, while we're at it, what about:
* Also having it search `$MINICONDA/share/openmm/forcefields/`? I think we can use `pkg_resources.resource_filename()` to get the `$MINICONDA` path.
* Allowing the user to supplement (or override?) the search path with an environment variable, like `$OPENMM_FORCEFIELD_PATH`?

**Comment by peastman** [comment]
I'd rather not.  This isn't meant to be a free for all where users indiscriminately add files to the default search path.  That could easily get out of hand and create problems where people aren't sure where their files are loaded from.

**Pull request merged**